source/distributed/shared_tria.cc

```diff
@@ -76,14 +103,85 @@ namespace parallel
                   active_halo_layer.find(cell) == active_halo_layer.end())
                 cell->set_subdomain_id(numbers::artificial_subdomain_id);
             }
+
+          // loop over all cells in multigrid hierarchy and mark artificial:
+          if (settings & construct_multigrid_hierarchy)
```

Comment:
i would add assert because in the current state of code `construct_multigrid_hierarchy` won't be used in full unless you have `allow_artificial_cells==true`.